spec/support/job_shared_examples.rb

```diff
@@ -29,13 +29,31 @@
     let(:job_transaction) { described_class.initialize_job(existing_container) }
     context 'perform_now' do
       include_context 'tracking job', :job_transaction
-      it {
-        expect(existing_container).to be_persisted
-        mocked_proxy = double()
-        expect(mocked_proxy).to receive(:update_document)
-        expect(existing_container).to receive(:__elasticsearch__).and_return(mocked_proxy)
-        ElasticsearchIndexJob.perform_now(job_transaction, existing_container, update: true)
-      }
+
+      context 'index exists' do
+        it {
+          expect(existing_container).to be_persisted
+          mocked_proxy = double()
+          expect(mocked_proxy).to receive(:update_document)
+            .with(ignore: 404).and_return(true)
```

Comment:
I think it would be better do to this as an integration test against Elasticsearch. This smells of "don't mock what you don't own".